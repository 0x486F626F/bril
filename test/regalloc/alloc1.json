{
  "functions": [
    {
      "instrs": [
        {
          "dest": "vb",
          "op": "const",
          "type": "int",
          "value": 5
        },
        {
          "dest": "vc",
          "op": "const",
          "type": "int",
          "value": 3
        },
        {
          "args": [
            "vb",
            "vc"
          ],
          "dest": "va",
          "op": "add",
          "type": "int"
        },
        {
          "args": [
            "va"
          ],
          "dest": "vd",
          "op": "id",
          "type": "int"
        },
        {
          "args": [
            "va",
            "vd"
          ],
          "dest": "vc",
          "op": "add",
          "type": "int"
        },
        {
          "args": [
            "vc"
          ],
          "op": "print"
        }
      ],
      "name": "main"
    }
  ]
}
