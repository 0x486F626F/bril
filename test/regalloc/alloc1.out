main {
	r_02: int = const 5;
	r_01: int = const 3;
	vb: int = id r_02
	va: int = id r_02;
	r_02: int = id vb;
	r_02: int = r_02 add r_01;
	vc: int = id r_01
	r_01: int = id r_02;
	vd: int = id r_01
	vc: int = id r_01;
	r_01: int = id vd;
	r_01: int = r_02 add r_01;
	print r_01;
}
